### DISCUSSION TITLE
Meta: Not Rocket Science

### DISCUSSION THREAD

**Discussion by RossSmyth** [helix-editor/helix#10093] [Ideas]
I started using Helix recently and build it from the top of main every Monday morning. While looking at what interesting commits came about this past week I noticed that Helix seems to run its tests against each branch, rather than the merged branch. I think it is a good idea to do the latter, which is often called the "[Not Rocket Science](https://graydon2.dreamwidth.org/1597.html)" rule. This can help catch bugs that running just PR CI will miss.

As a stop-gap using [GitHub Merge Queues](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue) would work, but they do have some [nice features that are missing](https://matklad.github.io/2023/06/18/GitHub-merge-queue.html). [bors-ng/bors-ng](https://github.com/bors-ng/bors-ng) is a good implementation but the public instance is being phased out since GH Merge Queues exist. Self-hosting is an option though. There is also Bors that rust-lang uses, but I wouldn't recommend that as it is old, has known bugs, and the codebase is a known pita to change. [It is being rewritten though](https://github.com/rust-lang/bors).

**Comment by pascalkuthe** [comment]
We tried tod witch to mergequeues but it requires that every pr is rebased on master (so that it has the mergequeue in its ci configuration). We put off switching to mergequeues until that is fixed

**Comment by RossSmyth** [comment]
I've not used GH Merge Queues personally, but does that matter? Can you enable them now so that new PRs will have it, then for old PRs they can rebase when they are ready to be merged? It's just sketchy to me to see a relatively large project have many open PRs and not have this strategy employed cause I've seen projects get funny compositional bugs without it.

**Reply by pascalkuthe** [reply]
Ot malediven ot impossible to merge old prs they get Stuhl in the merge queue forever
